pkg/service/healthserver/healthserver.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package healthserver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/cilium/cilium/pkg/counter"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "service-healthserver")
+
+// ServiceName is the name and namespace of the service
+type ServiceName struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+// Service represents the object returned by the health server
+type Service struct {
+	Service        ServiceName `json:"service"`
+	LocalEndpoints int         `json:"localEndpoints"`
+}
+
+// NewService creates a new service
+func NewService(ns, name string, localEndpoints int) *Service {
+	return &Service{
+		Service: ServiceName{
+			Namespace: ns,
+			Name:      name,
+		},
+		LocalEndpoints: localEndpoints,
+	}
+}
+
+// healthServer is a running HTTP health server for a certain service
+type healthServer interface {
+	updateService(*Service)
+	shutdown()
+}
+
+// healthServerFactory creates a new HTTP health server, used for mocking
+type healthServerFactory interface {
```

Comment:
Maybe `s/healthServerFactory/healthHTTPServerFactory/` and `s/newHealthServer/newHealthHTTPServer` to better distinguish from `ServiceHealthServer` (the user of the factory method and the interface)?